### PR TITLE
Use faraday

### PR DESCRIPTION
**Pull request by prcongithub**
Replaces httparty with Faraday to support multi threaded systems.
HTTParty uses class variables to define host for connections.
These class variables were modified by fcm to make requests to multiple google api end points.
With Faraday we can use connection instance variables to connect to different api end points and not worry about multiple threads making api calls.

### CODE REVIEW

**Comment by kashif** [comment]
🙏thanks @prcongithub will review ASAP

**Comment by prcongithub** [comment]
Any update here?

**Comment by kashif** [comment]
thanks will push a new version out asap

File: fcm.gemspec [hunk 1]
```diff
@@ -21,5 +21,5 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
-  s.add_runtime_dependency('httparty', '~> 0.10', '>= 0.10.0')
+  s.add_runtime_dependency('faraday','0.15.4')
```
Comment (jibiel) [inline fcm.gemspec]: The dependency can be too strict for the apps that already using `faraday` in their `Gemfile`.

**Pull request merged**